crates/daphne-worker/src/durable/mod.rs

```diff
@@ -199,10 +199,19 @@ pub(crate) async fn state_set_if_not_exists<T: for<'a> Deserialize<'a> + Seriali
     Ok(None)
 }
 
-async fn req_parse<T: DeserializeOwned>(req: &mut Request) -> Result<T> {
+async fn req_parse<T>(req: &mut Request) -> Result<T>
+where
+    T: DurableRequestPayload,
+    // TODO(mendess): delete
+    T: serde::de::DeserializeOwned,
+{
     let bytes = req.bytes().await?;
-    bincode::deserialize(&bytes)
-        .map_err(|e| Error::RustError(format!("failed to deserialize bincode: {e:?}")))
+    T::decode_from_bytes(&bytes)
+        .map_err(|e| Error::RustError(e.to_string()))
+        .or_else(|_| {
+            bincode::deserialize(&bytes)
+                .map_err(|e| Error::RustError(format!("failed to deserialize bincode: {e:?}")))
+        })
```

Comment:
It might be helpful to temporarily log the cpanproto decode failure.

Reply:
I'll just include both in the error message instead.